### PR TITLE
#9960 only run visual regression tests after pr approval

### PR DESCRIPTION
**Pull request by tbrlpld**
# Description

This PR moves the "Percy CI" job from the general continuous integration workflow to a separate one. 
This new workflow is triggered whenever a pull request review is submitted. 
A conditional check in the beginning of the of the job checks that the submitted review was an approving one. 

If the submitted PR review was approving, then the Percy CI job is run as before. 

I have been able to test the basic triggering and the not running of the job when the PR review is only a comment. 
We can only check that the positive path of the conditional works once this PR is approved 😅. 

Also, because our Percy plan is currently out of data, I could not make sure that the screenshots properly show up in Percy. 
I was able to see the Percy CI status check showing up on the PR though. So I am hopeful this will work properly once the PR is approved and we have actual screenshot budget on Percy.

Related PRs/issues: #9960

# Checklist

**Tests**
- [x] Is the code I'm adding covered by tests? -> Kind of testing itself 🤔

**Changes in Models:**
- ~~[ ] Did I update or add new fake data?~~
- ~~[ ] Did I squash my migration?~~
- ~~[ ] Are my changes [backward-compatible](https://github.com/mozilla/foundation.mozilla.org/blob/main/docs/workflow.md#django-migrations-what-to-do-when-working-on-backward-incompatible-migrations). If not, did I schedule a deploy with the rest of the team?~~

**Documentation:**
- [x] Is my code documented?
- [x] Did I update the READMEs or wagtail documentation?

### CODE REVIEW

**Review by tbrlpld — COMMENTED** [review]
Test

**Review by tbrlpld — COMMENTED** [review]
Test again

**Comment by tbrlpld** [comment]
The triggers seem to work. Just can't properly test right now because we have already exceeded our Percy limit.

**Review by tbrlpld — COMMENTED** [review]
Test with active conditional check

**Review by danielfmiranda — APPROVED** [review]
Hi @tbrlpld looks good to me! Thanks for taking this on I am excited to get this working as I am sure it is going to be very helpful.

Approved 🙌

**Review by mmmavis — APPROVED** [review]
Code looks good to me. Hopefully things work perfectly after PR gets merged. Thanks for the work @tbrlpld

**Review by fessehaye — APPROVED** [review]

File: .github/workflows/visual-regression-testing.yml [hunk 1]
```diff
@@ -3,11 +3,13 @@ name: Visual Regression Tests
...
     types: [submitted]
-
+  push:
+    branches:
+      - 'main'
 jobs:
   approved:
     name: Percy CI
-    if: github.event.review.state == 'approved'
+    if: github.event_name == 'push' || github.event.review.state == 'approved'
```
Comment (mmmavis) [inline .github/workflows/visual-regression-testing.yml:12]: I don't know yml format for GitHub workflow much.

Wondering if we need to update the above so they are two "jobs" triggering the same set of tasks.

Condition for Job 1) on push to `main` only
Condition for Job 2) when any PR gets a "approved" review

The first part of line 12 looks like push to any branch would trigger Percy?

Comment (tbrlpld) [reply .github/workflows/visual-regression-testing.yml:12]: The condition will pass for any push but the workflow is only triggered for push to main.

Comment (tbrlpld) [reply .github/workflows/visual-regression-testing.yml:12]: I can add an additional `&&` condition just make that clearer for readability.

Comment (tbrlpld) [reply .github/workflows/visual-regression-testing.yml:12]: I would not want to duplicate the job, because then we need to keep that all in sync.

Comment (mmmavis) [reply .github/workflows/visual-regression-testing.yml:12]: > I can add an additional `&&` condition just make that clearer for readability. 

@tbrlpld  yes please. Thank you!

Comment (tbrlpld) [reply .github/workflows/visual-regression-testing.yml]: @mmmavis I have updated the condition.

**Review by mmmavis — APPROVED** [review]
Thanks for making the conditions clearer to read!

**Review by fessehaye — APPROVED** [review]

**Pull request merged**